src/Security/MemberAuthenticator/Authenticator.php

```diff
@@ -0,0 +1,193 @@
+<?php
+
+namespace SilverStripe\Security\MemberAuthenticator;
+
+use SilverStripe\Control\Controller;
+use SilverStripe\Control\Session;
+use SilverStripe\ORM\ValidationResult;
+use InvalidArgumentException;
+use SilverStripe\Security\Authenticator as BaseAuthenticator;
+use SilverStripe\Security\Security;
+use SilverStripe\Security\Member;
+use SilverStripe\Security\LoginAttempt;
+
+/**
+ * Authenticator for the default "member" method
+ *
+ * @author Markus Lanthaler <markus@silverstripe.com>
+ */
+class Authenticator implements BaseAuthenticator
+{
+
+    public function supportedServices()
+    {
+        // Bitwise-OR of all the supported services, to make a bitmask
+        return BaseAuthenticator::LOGIN | BaseAuthenticator::LOGOUT | BaseAuthenticator::CHANGE_PASSWORD
+            | BaseAuthenticator::RESET_PASSWORD;
+    }
+
+    /**
+     * @inherit
+     */
+    public function authenticate($data, &$message)
+    {
+        $success = null;
+
+        // Find authenticated member
+        $member = $this->authenticateMember($data, $message, $success);
+
+        // Optionally record every login attempt as a {@link LoginAttempt} object
+        $this->recordLoginAttempt($data, $member, $success);
+
+        if ($member) {
+            Session::clear('BackURL');
+        }
+
+        return $success ? $member : null;
+    }
+
+    /**
+     * Attempt to find and authenticate member if possible from the given data
+     *
+     * @param array $data Form submitted data
+     * @param $message
+     * @param bool &$success Success flag
+     * @param null|Member $member If the parent method already identified the member, it can be passed in
+     * @return Member Found member, regardless of successful login
+     */
+    protected function authenticateMember($data, &$message, &$success, $member = null)
```

Comment:
It is in CMSAuthenticator, where the member is fetched from the TempID and passed in to further authenticate.